### PR TITLE
allow users to use query builder while in viewer mode

### LINKED ISSUE #297: RFC: Viewer mode improvements
## 💬 Request for Comments

Viewer mode allows user to enter a read-only workspace that corresponds to the current HEAD of the project, we should come up with a set of features that we should allow in this mode. Modifications to models should be blocked. This can set us up on the right path should we update review mode to also allow users to interact with the read-only workspace.

### Implementation Plan
- [ ] Allow seeing project configuration
- [ ] Allow hackermode/text-mode
- [ ] Allow opening model loader
- [ ] Allow using query builder

### PR DESCRIPTION
**Pull request by umarphaarook**
| Q                       | A                                |
| ----------------------- | -------------------------------- |
| Fixed issues?           | Related to #297  |
| Tests added + passed?   | No                              |
| Changeset added?        | Yes                              |
| Documentation PR link   |                                  |
| Any dependency changes? | No                               |

### CODE REVIEW

File: .changeset/strange-meals-whisper.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@finos/legend-studio': patch
+---
+
+Allow users to use query builder while in viewer mode.
```
Comment (MauricioUyaguari) [inline .changeset/strange-meals-whisper.md:5]: Enable query builder in viewer mode.

File: packages/legend-studio/src/components/editor/side-bar/Explorer.tsx [hunk 2]
```diff
@@ -190,14 +191,14 @@ const ExplorerContextMenu = observer(
...
-        )}
-        {node && (
-          <MenuContentItem onClick={openElementInViewerMode}>
-            View in Project
-          </MenuContentItem>
+        {isNotInViewerMode && <MenuContentItem>Rename (WIP)</MenuContentItem>}
+        {isNotInViewerMode && node && (
+          <>
+            <MenuContentItem onClick={deleteElement}>Delete</MenuContentItem>
+            <MenuContentItem onClick={openElementInViewerMode}>
```
Comment (MauricioUyaguari) [inline packages/legend-studio/src/components/editor/side-bar/Explorer.tsx:198]: can we enable 'copy link element' in viewer mode. users would still want to share the element link.

**Comment by akphi** [comment]
Could you fix the description to say `Related to https://github.com/finos/legend-studio/issues/297` instead of using `Fixes`. There are other issues in https://github.com/finos/legend-studio/issues/297 that this PR will not solve. Using `fixes` will cause the issue to be closed when we merge this PR

File: packages/legend-studio/src/components/editor/side-bar/Explorer.tsx (hunk 2 above)
Comment (umarphaarook) [reply packages/legend-studio/src/components/editor/side-bar/Explorer.tsx:198]: `Copy link` works in viewer mode, I can confirm.

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**